controllers/cloud.redhat.com/providers/web/impl.go

```diff
@@ -68,17 +91,659 @@ func (web *webProvider) makeService(deployment *crd.Deployment, app *crd.ClowdAp
+	WebOrigins                []string       `json:"webOrigins"`
+	ProtocolMappers           []mapperStruct `json:"protocolMappers"`
+	DirectAccessGrantsEnabled bool           `json:"directAccessGrantsEnabled"`
+}
+
+func (k *KeyCloakClient) createClient(realmName string, clientName string) error {
+	headers := map[string]string{
+		"Content-Type": "application/json",
+	}
+
+	postObj := clientStruct{
+		ClientId:                  clientName,
+		Enabled:                   true,
+		BearerOnly:                false,
+		PublicClient:              true,
+		RedirectUris:              []string{"url"},
+		WebOrigins:                []string{"url"},
+		DirectAccessGrantsEnabled: true,
+		ProtocolMappers: []mapperStruct{
+			createMapper("account_number", "String"),
+			createMapper("account_id", "String"),
+			createMapper("org_id", "String"),
+			createMapper("username", "String"),
+			createMapper("email", "String"),
+			createMapper("first_name", "String"),
+			createMapper("last_name", "String"),
+			createMapper("is_org_admin", "boolean"),
+			createMapper("is_internal", "boolean"),
+			createMapper("is_active", "boolean"),
+			createMapper("entitlements", "String"),
+		},
+	}
+
+	b, err := json.Marshal(postObj)
+
+	if err != nil {
+		return err
+	}
+
+	resp, err := k.Post(
+		fmt.Sprintf("/auth/admin/realms/%s/clients", realmName),
+		string(b), headers,
+	)
+
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("\n\n%v\n\n", resp)
+	v, _ := ioutil.ReadAll(resp.Body)
+	fmt.Printf("\n\n%s\n\n", string(v))
+	return nil
+}
+
+func (k *KeyCloakClient) createUser(realmName string, user *createUserStruct) error {
+	headers := map[string]string{
+		"Content-Type": "application/json",
+	}
+
+	b, err := json.Marshal(user)
+
+	if err != nil {
+		return err
+	}
+
+	resp, err := k.Post(
+		fmt.Sprintf("/auth/admin/realms/%s/users", realmName),
+		string(b), headers,
+	)
+
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("\n\n%v\n\n", resp)
+	v, _ := ioutil.ReadAll(resp.Body)
+	fmt.Printf("\n\n%s\n\n", string(v))
+	return nil
+}
+
+func (m *localWebProvider) configureKeycloak() error {
+	s := &core.Service{}
+	if err := m.Cache.Get(WebKeycloakService, s); err != nil {
+		return err
+	}
+
+	client, err := NewKeyCloakClient(fmt.Sprintf("http://%s.%s.svc:8080", s.Name, s.Namespace), "admin", "admin", m.Ctx)
+
+	if err != nil {
+		return err
+	}
+
+	exists, err := client.doesRealmExist("redhat-external")
+
+	if err != nil {
+		return err
+	}
+
+	if !exists {
+		err := client.createRealm("redhat-external")
+		if err != nil {
+			return err
+		}
+	}
+
+	exists, err = client.doesClientExist("redhat-external", "cloud-services")
+
+	if err != nil {
+		return err
+	}
+
+	if !exists {
+		err := client.createClient("redhat-external", "cloud-services")
+		if err != nil {
+			return err
+		}
+	}
+
+	exists, err = client.doesUserExist("redhat-external", "jdoe")
+
+	if err != nil {
+		return err
+	}
+
+	user := &createUserStruct{
```

Comment:
move into `if` block or move to a different fn

Reply:
done